File: py/ds/stack.py
arr = []
arr = [0]*10

size=0

def push(data):
    global size
    arr[size] = data
    size = size+1
    
def pop():
    global size
    arr[size]=0
    size = size-1
    
def printArr():
    for i in range(0,size):
        print arr[i]
 
push(1)
push(2)
push(3)
push(4)

pop()

#print size
printArr()

File: ruby/ds/stack.rb
$arr = Array.new()

def push (data)
    $arr.push(data)
end
    
def pop()
    $arr.pop()
end
    
def printArr()
    for a in $arr
        puts "#{a}"
    end
end
 
push(1)
push(2)
push(3)
push(4)

pop()

puts "size is: #{$arr.length()}"
printArr()